ww3tools/modelBuoy_collocation.py

```diff
@@ -883,7 +883,10 @@ def unzip_and_untar(gz_file, output_dir):
 
         ncfile = nc.Dataset(f'WW3.{model_name}Buoy{ftag}_{initime}to{fintime}.nc', "w", format=fnetcdf)
         print(f"Model Name: {model_name}, Tag: {ftag}, Start Time: {initime}, End Time: {fintime}")
-        ncfile.history = "Matchups of WAVEWATCHIII point output (table) and NDBC and Copernicus Buoys. Total of " + repr(bhs[bhs > 0.].shape[0]) + " observations or pairs model/observation."
+        num_time_points = nmhs.shape[2]  # Total number of time points
+        num_buoy_points = bhs.shape[0]  # Total number of buoy points
+        total_observations = num_time_points * num_buoy_points
+        ncfile.history = "Matchups of WAVEWATCHIII point output (table) and NDBC and Copernicus Buoys. Total of " + str(total_observations) + " observations or pairs model/observation."
```

Comment:
I am not sure about this change. We have to actually calculate the matchups. similar to what was in the code before. What you have added here assumes that for each station and model time step we have buoy outputs which is not always correct, that is the maximum matchups we could get. The output you had shared had "Total of 4417048 observations or pairs model/observation." which was way beyond the number I was expecting for 166 buoys and 57 timesteps. Please take a closer look and check this.